cardillo/math/rotations.py

```diff
@@ -554,12 +554,11 @@ def Exp_SO3_quat(P, normalize=True):
     Nuetzi2016: https://www.research-collection.ethz.ch/handle/20.500.11850/117165
     """
     p0, p = np.array_split(P, [1])
-    p_tilde = ax2skew(p)
     if normalize:
         P2 = P @ P
-        return eye3 + (2 / P2) * (p0 * p_tilde + p_tilde @ p_tilde)
+        return eye3 + (2 / P2) * (p0 * ax2skew(p) + ax2skew_squared(p))
```

Comment:
Here we can multiply p with p0 before passing it to the ax2skew function

Reply:
@JonasBreuling what do you think? you meant that it would lose overview of the code if it is optimized too much.

Reply:
Agree. Add a unit-test and make the "derivation" starting from the form found in literature to the implemented form in the comments.